metadata-ingestion/src/datahub/ingestion/api/source.py

```diff
@@ -54,7 +55,9 @@ def get_records(self, workunit: WorkUnitType) -> Iterable[RecordEnvelope]:
 # See https://github.com/python/mypy/issues/5374 for why we suppress this mypy error.
 @dataclass  # type: ignore[misc]
 class Source(Closeable, metaclass=ABCMeta):
-    ctx: PipelineContext
+    def __init__(self, ctx: PipelineContext) -> None:
+        self.report: SourceReport = SourceReport()
```

Comment:
I don't think this is the right thing to do. Instead, report should be passed as an argument of __init__ via a generic typing.TypeVar bound to SourceReport. This will incur a huge refactoring across and we should do it as part of a different PR.